woocommerce/woocommerce
# Cart Block: Reduce prop drilling in shipping calculator by introducing context

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Shipping calculator refactor split from https://github.com/woocommerce/woocommerce/pull/52313/files#diff-c628d2fab853de36d70da769aefdb76210fb6bab3635a82ed07eaf7700fb3957 which introduces context to avoid prop drilling. This context takes state, ID, and if the calculator should be enabled.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add something to the cart
2. Go to the cart block page
3. Toggle on the shipping calculator open
4. Fill out the form and update
5. Calculator closes and rates update
6. Confirm no regressions on checkout page (no calculator is shown there)



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Refactor to shipping calculator in cart block. No functionality changes.

</details>
